core/dbt/cli/main.py

```diff
@@ -121,11 +121,18 @@ def invoke(self, args: List[str], **kwargs) -> dbtRunnerResult:
 def global_flags(func):
     @p.cache_selected_only
     @p.debug
+    @p.defer
+    @p.deprecated_defer
+    @p.defer_state
+    @p.deprecated_favor_state
     @p.deprecated_print
+    @p.deprecated_state
     @p.enable_legacy_logger
     @p.fail_fast
+    @p.favor_state
```

Comment:
@jtcohen6, re: https://github.com/dbt-labs/dbt-core/pull/9040#issuecomment-1889709518

is this just a matter of adding indirect_selection here? took it for a spin locally and it spun

Reply:
I'm guessing it would also involve removing `@p.indirect_selection` from each of the sub commands.